test/e2e/idle_connection_test.go

```diff
@@ -0,0 +1,778 @@
+		return nil, fmt.Errorf("error waiting for route to be admitted: %w", err)
+	}
+
+	t.Logf("Route %s/%s admitted by ingresscontroller %s", route.Namespace, route.Name, ic.Name)
+
+	if len(tc.deployments) != 2 {
+		return nil, fmt.Errorf("expected 2 deployments, but got %d", len(tc.deployments))
+	}
+
+	if len(tc.services) != 2 {
+		return nil, fmt.Errorf("expected 2 services, but got %d", len(tc.services))
+	}
+
+	if len(tc.pods) != 2 {
+		return nil, fmt.Errorf("expected 2 pods, but got %d", len(tc.pods))
+	}
+
+	tc.routeName = types.NamespacedName{Namespace: route.Namespace, Name: route.Name}
+
+	return tc, nil
+}
+
+func idleConnectionCreateBackendService(ctx context.Context, t *testing.T, ns *corev1.Namespace, tc *idleConnectionTestConfig, index int, serverResponse, image string) error {
+	svc, err := idleConnectionCreateService(ctx, ns.Name, index)
+	if err != nil {
+		return fmt.Errorf("failed to create service %d: %w", index, err)
+	}
+	tc.services = append(tc.services, svc)
+
+	deployment, err := idleConnectionCreateDeployment(ctx, ns.Name, index, serverResponse, image)
+	if err != nil {
+		return fmt.Errorf("failed to create deployment %d: %w", index, err)
+	}
+	tc.deployments = append(tc.deployments, deployment)
+
+	if err := waitForDeploymentComplete(t, kclient, deployment, 2*time.Minute); err != nil {
+		return fmt.Errorf("deployment %d is not ready: %w", index, err)
+	}
+
+	return nil
+}
+
+func idleConnectionCreateDeployment(ctx context.Context, namespace string, serviceNumber int, serverResponse, image string) (*appsv1.Deployment, error) {
+	name := fmt.Sprintf("web-server-%d", serviceNumber)
+	secretName := fmt.Sprintf("serving-cert-%s-%s", namespace, name)
+
+	selectorLabels := map[string]string{
+		"app":      "web-server",
+		"instance": fmt.Sprintf("%d", serviceNumber),
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    selectorLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.To[int32](1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selectorLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: selectorLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            name,
+							Image:           image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         []string{"/usr/bin/ingress-operator"},
+							Args:            []string{"serve-http2-test-server"},
+							Ports: []corev1.ContainerPort{
+								{Name: "http", ContainerPort: 8080},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "CUSTOM_RESPONSE", Value: serverResponse},
+								{Name: "PORT", Value: "8080"},
+								{Name: "TLS_CERT", Value: "/etc/serving-cert/tls.crt"},
+								{Name: "TLS_KEY", Value: "/etc/serving-cert/tls.key"},
```

Comment:
Why do we need all the hustle with the serving cert secret if at the end we hit the plain http port?

Reply:
Because I rely on the server in test/http2 which requires certificates for the server that listens on port https.

Reply:
I see now. Does it make sense to make the listening on HTTPS port conditional (== don't listen on HTTPS is not provided)?

Reply:
I don't want to change existing tests that rely on the test http2 server. Nor do I really want to change anything that would make the backport more complicated. This e2e test has to go back to 4.14.

Reply:
If we want to modify the test/http2 server then that can be pursued as as separate PR.